types/sql.d.ts

```diff
@@ -1,7 +1,40 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License.
 
-import { FunctionInput, FunctionOutput } from './index';
+import { FunctionInput, FunctionOptions, FunctionOutput, FunctionResult, FunctionTrigger } from './index';
+import { InvocationContext } from './InvocationContext';
+
+export type SqlHandler = (changes: SqlChange[], context: InvocationContext) => FunctionResult;
+
+export interface SqlFunctionOptions extends SqlTriggerOptions, Partial<FunctionOptions> {
+    handler: SqlHandler;
+
+    trigger?: SqlTrigger;
+}
+
+export interface SqlTriggerOptions {
+    /**
+     * The name of the table monitored by the trigger.
+     */
+    tableName: string;
+
+    /**
+     * An app setting (or environment variable) with the connection string for the database containing the table monitored for changes
+     */
+    connectionStringSetting: string;
+}
+export type SqlTrigger = FunctionTrigger & SqlTriggerOptions;
+
+export interface SqlChange {
+    Item: unknown;
```

Comment:
I wish these were lowercase, and I'm also not a fan of "Operation" using an enum, but this is what the SQL trigger binding gives us, so that's what the types have to reflect (unless we decide to automatically convert the trigger input, but we don't do that for any other trigger input and I didn't think it was worth it)